src/org/labkey/test/pages/core/login/LoginConfigurePage.java

```diff
@@ -20,61 +23,130 @@ public LoginConfigurePage(WebDriver driver)
+        return this;
+    }
+
+    public boolean getAllowEditEmail()
+    {
+        return elementCache().allowUserEmailEditCheckbox.get();
+    }
+
+    public LoginConfigurePage setAutoCreate(boolean enable)
+    {
+        elementCache().autoCreateCheckBox.set(enable);
+        return this;
+    }
+
+    public boolean getAutoCreateEnabled()
+    {
+        return elementCache().autoCreateCheckBox.get();
+    }
+
+    public LoginConfigurePage togglePrimaryConfiguration()
+    {
+        if (!isPrimarySelected())
+            elementCache().panelTab1.click();
+        waitFor(()-> isPrimarySelected(), 1000);
+        return this;
+    }
+
+    public LoginConfigRow getPrimaryConfigurationRow(String description)
+    {
+        return new LoginConfigRow.LoginConfigRowFinder(getDriver()).withDescription(description)
+                .waitFor(elementCache().tabPane1);
+    }
+
+    public List<LoginConfigRow> getPrimaryConfigurations()
+    {
+        togglePrimaryConfiguration();
+        return new LoginConfigRow.LoginConfigRowFinder(getDriver()).findAll(elementCache().tabPane1);
+    }
+
+    public LoginConfigurePage toggleSecondaryConfiguration()
+    {
+        if (!isSecondarySelected())
+            elementCache().panelTab2.click();
+        waitFor(()-> isSecondarySelected(), 1000);
+        return this;
+    }
+
+    public List<LoginConfigRow> getSecondaryConfigurations()
+    {
+        toggleSecondaryConfiguration();
+        return new LoginConfigRow.LoginConfigRowFinder(getDriver()).findAll(elementCache().tabPane2);
+    }
 
-        WebElement row = Locator.xpath("//tbody/tr").containing(description).findElement(elementCache().tableElement);
-        clickAndWait(Locator.tagWithClass("a", "labkey-text-link").withText("edit").findElement(row));
+    public LoginConfigRow getSecondaryConfigurationRow(String description)
+    {
+        toggleSecondaryConfiguration();
+        return new LoginConfigRow.LoginConfigRowFinder(getDriver())
+                .withDescription(description).waitFor(elementCache().tabPane2);
+    }
+
+    public LoginConfigurePage removeConfiguration(String description)      // assumes for now we're doing primary only
+    {
+        new LoginConfigRow.LoginConfigRowFinder(getDriver()).withDescription(description).waitFor()
+                .clickDelete();
+        return this;
+    }
 
-        return page;
+    public <D extends AuthDialogBase> D clickEditConfiguration(String description, AuthenticationProvider<D> authenticationProvider)
+    {
+        LoginConfigRow row = new LoginConfigRow.LoginConfigRowFinder(getDriver()).withDescription(description).waitFor(getDriver());
+        return row.clickEdit(authenticationProvider);
     }
 
-    public ShowAdminPage clickDone()
+    public ShowAdminPage clickSaveAndFinish()
     {
-        clickAndWait(elementCache().doneBtn());
+        shortWait().until(ExpectedConditions.elementToBeClickable(elementCache().saveAndFinishBtn()));
```

Comment:
Does the save button get disabled in a way that this will notice?

Reply:
not 100% sure.
Added this on the chance that it might be helpful; we may want to come up with a consistent means of knowing if a button is in a disabled state